torch/nn/modules/activation.py

```diff
@@ -868,7 +868,10 @@ class MultiheadAttention(Module):
     bias_k: Optional[torch.Tensor]
     bias_v: Optional[torch.Tensor]
 
-    def __init__(self, embed_dim, num_heads, dropout=0., bias=True, add_bias_kv=False, add_zero_attn=False, kdim=None, vdim=None):
+    def __init__(self, embed_dim, num_heads, dropout=0., bias=True, add_bias_kv=False,
+                 add_zero_attn=False, kdim=None, vdim=None,
+                 device='cpu', dtype=torch.get_default_dtype()) -> None:
```

Comment:
Representative sample 1: standard case, nothing special going on. Most module updates look like this.

Reply:
Joel, I think it's better to do `device=None, dtype=None`. It is shorter, doesn't require you to run a function inside the default argument creation, and leaves the possibility of adding a factory context manager in the future